### PR TITLE
update doc for join command

### PR DESCRIPTION
**Pull request by ycyaoxdu**
Signed-off-by: ycyaoxdu <yaoyuchen0626@163.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,14 +86,20 @@ Initialize the hub by deploying the hub side resources to manage clusters.
...
 ### join
 
 Install the agent on the spoke.
 
-`clusteradm join --hub-token <token> --hub-apiserver <hub_apiserver_url> --cluster-name c1`
+`clusteradm join --hub-token <token> --hub-apiserver <hub_apiserver_url> --cluster-name c1 [--ca-file <path-to-ca-file>] [--force-internal-endpoint-lookup]`
 
 it returns the command line to launch on the hub the accept the spoke onboarding.
 
+> NOTE: While clusteradm could fetch any ca data from given `--hub-apiserver`, then `--ca-file` flag should be set.
```
Comment (nirs) [inline README.md:99]: Not clear what do you mean. Why --ca-file should be used and in which case?

File: README.md [hunk 2]
```diff
@@ -86,14 +86,20 @@ Initialize the hub by deploying the hub side resources to manage clusters.
 
 it returns the command line to launch on the spoke to join the hub.
 
+> NOTE: Do not run init command against a multicluster-controlplane instance. It is already a initialized hub onstart. Instead, Using `clusteradm get token --use-bootstrap-token` to get the join command
```
[the remark this answers is not in the document]
Comment (ycyaoxdu) [reply README.md]: done

File: README.md (hunk 1 above)
Comment (ycyaoxdu) [reply README.md]: done

File: README.md [hunk 3]
```diff
@@ -85,21 +85,25 @@ Initialize the hub by deploying the hub side resources to manage clusters.
...
 ### join
 
 Install the agent on the spoke.
 
-`clusteradm join --hub-token <token> --hub-apiserver <hub_apiserver_url> --cluster-name c1`
+`clusteradm join --hub-token <token> --hub-apiserver <hub_apiserver_url> --cluster-name c1 [--ca-file <path-to-ca-file>] [--force-internal-endpoint-lookup]`
 
 it returns the command line to launch on the hub the accept the spoke onboarding.
 
+> NOTE: While clusteradm could not fetch any CA data from given `--hub-apiserver`, then `--ca-file` flag should be set to provide a valid CA.
```
Comment (qiujian16) [inline README.md:103]: we probably add more description here. The ca data is fetch from xx secret kube-public namespace it is not set.

Comment (ycyaoxdu) [reply README.md]: done

File: README.md [hunk 4]
```diff
@@ -85,21 +85,25 @@ Initialize the hub by deploying the hub side resources to manage clusters.
...
 ### join
 
 Install the agent on the spoke.
 
-`clusteradm join --hub-token <token> --hub-apiserver <hub_apiserver_url> --cluster-name c1`
+`clusteradm join --hub-token <token> --hub-apiserver <hub_apiserver_url> --cluster-name c1 [--ca-file <path-to-ca-file>] [--force-internal-endpoint-lookup]`
 
 it returns the command line to launch on the hub the accept the spoke onboarding.
 
+> NOTE: The `--ca-file` flag is used to provide a valid CA for hub. The ca data is fetch from cluster-info configmap in kube-public namespace of the hub cluster if it is not set, then from kube-root-ca.crt configmap in kube-public if cluster-info configmap also not exist.
```
Comment (qiujian16) [inline README.md:103]:
```suggestion
> NOTE: The `--ca-file` flag is used to provide a valid CA for hub. The ca data is fetched from cluster-info configmap in kube-public namespace of the hub cluster if it is not set, then from kube-root-ca.crt configmap in kube-public namespace if the cluster-info configmap does not exist.
```

**Pull request merged**